src/liger_kernel/ops/fused_linear_cross_entropy.py

```diff
@@ -59,6 +59,10 @@ def fused_linear_cross_entropy_forward(
         logits_chunk = _input_chunk @ weight.t()  # chunk_size x V
         if bias is not None:
             logits_chunk = logits_chunk + bias
+        if logits_chunk.dtype != weight.dtype:
```

Comment:
Is dtype comparison necessary? According to aten source code, I think [.to(dtype)](https://github.com/pytorch/pytorch/blob/d8c3900d80c4f170759775fd2500d08f3ee83166/aten/src/ATen/native/TensorConversions.cpp#L417) already does a dtype check implicitly.

Reply:
Yes, you are right! After reading the source code you provided, conducting some small experiments, and profiling the behavior of PyTorch, it is confirmed that it performs a dtype check implicitly and bypasses the casting operation if the dtype is already the same.

Here is an example of a bfloat16 tensor attempting `.to(dtype=torch.float32)`:
![image](https://github.com/user-attachments/assets/22f53d3b-455a-4172-a6f7-8787894822d6)

And here is an example of a float32 tensor attempting `.to(dtype=torch.float32)`:
![image](https://github.com/user-attachments/assets/8cde8399-f63d-46c7-a2a0-5107e6f9825f)

I will fix this by removing the unnecessary comparison.

Reply:
Hi @Tcc0403, thanks for the review! I have merged the latest main branch and made a new commit to remove the unnecessary `dtype` comparison. I also re-ran `make test`, `make checkstyle`, and `make test-convergence`, and all tests have passed as before.

Reply:
Just curious how it would affect the speed and memory if we move type conversion into `addmm()`. https://github.com/linkedin/Liger-Kernel/pull/502/files#diff-d7a61f0e3b11dccec3ffab797033897bbc73d69abff5538f6ff11da850e38f63R120
```python
if grad_weight is not None:
            torch.addmm(
                input=grad_weight,
                mat1=logits_chunk.t().to(_input_chunk), # here
                mat2=_input_chunk,
                out=grad_weight,
                alpha=alpha,
                beta=1.0,
            )
```

Reply:
That looks interesting! I can conduct some experiments to profile the speed and memory.